src/service/viewport-impl.js

```diff
@@ -290,10 +296,11 @@ export class Viewport {
 
   /**
    * Returns the size of the viewport.
+   * @param {SizeReportingFidelity} fidelity
    * @return {!{width: number, height: number}}
    */
-  getSize() {
-    if (this.size_) {
+  getSize(fidelity = SizeReportingFidelity.LOW) {
+    if (this.size_ && fidelity == SizeReportingFidelity.LOW) {
```

Comment:
Correct. In that case, I believe the problem was that the resize event is triggered too late. Might make sense to keep fidelity then. @aghassemi ?